holmes/plugins/toolsets/database/database.py

```
@@ -107,6 +111,86 @@ def _detect_subtype(connection_url: str) -> DatabaseSubtype:
+
+def _execute_clickhouse_http(
+    base_url: str,
+    database: str,
+    auth_header: Optional[str],
+    sql: str,
+    effective_limit: int,
+    timeout_seconds: int = 60,
+) -> Dict[str, Any]:
+    """Execute a read-only ClickHouse query via HTTP API with JSONEachRow."""
+    query_url = f"{base_url}/?database={database}&default_format=JSONEachRow"
+    req = Request(query_url, data=sql.encode("utf-8"), method="POST")
```

Ruff (0.15.13)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)